src/sentry/lang/javascript/processor.py

```diff
@@ -780,11 +707,15 @@ def _get_debug_id_artifact_bundle_entry(self, debug_id, source_file_type):
         """
         # TODO: in the future we would like to load all the artifact bundles that are connected to the projects
         #  we have permissions on not all the bundles.
-        return DebugIdArtifactBundle.objects.filter(
-            organization_id=self.organization.id,
-            debug_id=debug_id,
-            source_file_type=source_file_type.value if source_file_type is not None else None,
-        ).select_related("artifact_bundle__file")[0]
+        return (
+            DebugIdArtifactBundle.objects.filter(
+                organization_id=self.organization.id,
+                debug_id=debug_id,
+                source_file_type=source_file_type.value if source_file_type is not None else None,
+            )
+            .order_by("-date_added")
+            .select_related("artifact_bundle__file")[:1][0]
```

Comment:
What's the point of slicing the array if we ask for the first element instantly afterwards anyway?

Reply:
It is because if you use slices django will put the `LIMIT` on the query.

Reply:
Then `.first()` is more understandable I guess.

Reply:
I had to debug the underlying SQL to make sure this is what was happening. I really don't like all of these abstractions on top of SQL.